### PR TITLE
sql selectors and new migrations

### PR DESCRIPTION
**Pull request by DenNegura**
migration and sql selects

### CODE REVIEW

File: homework-selects/sql_questions.sql [hunk 1]
```diff
@@ -1,57 +1,579 @@
...
+FROM employees
+    JOIN departments USING (department_id)
+    JOIN jobs USING (job_id)
+    WHERE department_name = 'Finance'
+	
 --19. all the information of an employee whose id is any of the number 134, 159 and 183.
+
+SELECT * 
+FROM employees 
+    WHERE employee_id IN (134, 159, 183);
```
Comment (rusnmaxim) [inline homework-selects/sql_questions.sql]: we need to print information only of one person

File: homework-selects/sql_questions.sql [hunk 2]
```diff
@@ -1,57 +1,579 @@
...
+    WHERE department_id IN (
+        SELECT department_id
+        FROM employees
+            WHERE first_name = 'Clara'
+    )
+    AND first_name <> 'Clara'
+	
 --24. the employee number and name( first name and last name ) for all employees who work in a department with any employee whose name contains a T.
+
+SELECT first_name || ' ' || last_name as "FULL NAME"
```
Comment (rusnmaxim) [inline homework-selects/sql_questions.sql]: you forgot the employee number

File: homework-selects/sql_questions.sql [hunk 3]
```diff
@@ -1,57 +1,579 @@
...
+    JOIN departments USING (department_id)
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+    )
+    AND department_name LIKE '%IT%'
+	
 --35. who earns more than Mr. Ozer.
+
+SELECT *
```
Comment (rusnmaxim) [inline homework-selects/sql_questions.sql:373]: this select is not working, fix it pls

File: homework-selects/sql_questions.sql [hunk 4]
```diff
@@ -1,57 +1,579 @@
...
+FROM employees
+    WHERE salary > (
+        SELECT salary
+        FROM employees
+            WHERE last_name || ' ' || first_name LIKE '%Ozer%';
+    )
+
 --36. which employees have a manager who works for a department based in the US.
+
+SELECT E.*
```
Comment (rusnmaxim) [inline homework-selects/sql_questions.sql:383]: incorect query, 122 manager doesn't work in US region, but I can see the employee with this manager

File: homework-selects/sql_questions.sql [hunk 5]
```diff
@@ -1,57 +1,579 @@
...
+FROM employees
+    WHERE salary > (
+        SELECT MAX(salary)
+        FROM employees
+        WHERE department_id = 40
+    );
+	
 --41. the department name and Id for all departments where they located, that Id is equal to the Id for the location where department number 30 is located.
+
+SELECT department_name,
```
Comment (rusnmaxim) [inline homework-selects/sql_questions.sql:449]: you need to print the locationId not department id

File: homework-selects/sql_questions.sql [hunk 6]
```diff
@@ -1,57 +1,579 @@
...
+FROM employees
+    WHERE department_id = (
+        SELECT department_id
+        FROM employees
+            WHERE employee_id = 201
+    );
+	
 --43. the first and last name, salary, and department ID for those employees whose salary is equal to the salary of the employee who works in that department which ID is 40.
+
+SELECT first_name,
```
Comment (rusnmaxim) [inline homework-selects/sql_questions.sql:473]: wrong query. reread '**the salary of the employee** who **works** in that **department which ID is 40**', not select all employees who work in department 40

File: src/main/resources/db/migration/V30000000000010__alter_table_location — копия.sql [hunk 7]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE locations
```
Comment (rusnmaxim) [inline src/main/resources/db/migration/V30000000000010__alter_table_location — копия.sql]: delete the copy

File: src/main/resources/db/migration/V30000000000011__create_triger_count_departments — копия.sql [hunk 8]
```diff
@@ -0,0 +1,18 @@
+CREATE OR REPLACE TRIGGER count_departments
+  AFTER INSERT OR DELETE ON departments
+  FOR EACH ROW
+DECLARE
+    loc_id locations.location_id%type;
```
Comment (rusnmaxim) [inline src/main/resources/db/migration/V30000000000011__create_triger_count_departments — копия.sql]: wow, didn't know thi is possible

File: src/main/resources/db/migration/V30000000000011__create_triger_count_departments — копия.sql [hunk 9]
```diff
@@ -0,0 +1,18 @@
+CREATE OR REPLACE TRIGGER count_departments
```
Comment (rusnmaxim) [inline src/main/resources/db/migration/V30000000000011__create_triger_count_departments — копия.sql]: again copy

File: src/main/resources/db/migration/V30000000000012__create_table_employment_logs — копия.sql [hunk 10]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE employment_logs
```
Comment (rusnmaxim) [inline src/main/resources/db/migration/V30000000000012__create_table_employment_logs — копия.sql]: file repeats

**Review by rusnmaxim — CHANGES_REQUESTED** [review]
please remove all the copies. also unfortunately I can't run the flyway on my pc since I have the problems with environment and I don't want to dive into it, so just send here the picture that it worked and logs

File: homework-selects/sql_questions.sql (hunk 1 above)
Comment (DenNegura) [reply homework-selects/sql_questions.sql]: Good, fixed by using FETCH

File: homework-selects/sql_questions.sql (hunk 2 above)
Comment (DenNegura) [reply homework-selects/sql_questions.sql]: Fixed. Sory!

File: homework-selects/sql_questions.sql (hunk 3 above)
Comment (DenNegura) [reply homework-selects/sql_questions.sql:373]: Fixed.

File: homework-selects/sql_questions.sql (hunk 4 above)
Comment (DenNegura) [reply homework-selects/sql_questions.sql:383]: Sory, by 122 manager works in US, in department 50

File: homework-selects/sql_questions.sql (hunk 5 above)
Comment (DenNegura) [reply homework-selects/sql_questions.sql:449]: Fixed, but condition is strange

**Comment by DenNegura** [comment]
Copies deleted. I don't know why it appeared.
image

File: src/main/resources/db/migration/V30000000000010__alter_table_location — копия.sql (hunk 7 above)
Comment (DenNegura) [reply src/main/resources/db/migration/V30000000000010__alter_table_location — копия.sql]: deleted

File: src/main/resources/db/migration/V30000000000011__create_triger_count_departments — копия.sql (hunk 9 above)
Comment (DenNegura) [reply src/main/resources/db/migration/V30000000000011__create_triger_count_departments — копия.sql]: deleted

File: src/main/resources/db/migration/V30000000000012__create_table_employment_logs — копия.sql (hunk 10 above)
Comment (DenNegura) [reply src/main/resources/db/migration/V30000000000012__create_table_employment_logs — копия.sql]: deleted

File: homework-selects/sql_questions.sql (hunk 5 above)
Comment (rusnmaxim) [reply homework-selects/sql_questions.sql:449]: at least there is description. sometimes there is posibility that you will recieve only task with name and no description, or description that you don;t understand fully. always contact the person who wrote it, or team members to inform that you don't get it properly

**Review by rusnmaxim — APPROVED** [review]

**Pull request merged**